apps/aragon-fundraising/app/src/views/MainView.js

```diff
@@ -61,8 +71,22 @@ export default () => {
     price: polledPrice,
     orderPanel,
     setOrderPanel,
+    userBondedBalance,
+    userDaiBalance,
+    userAntBalance,
   }
 
+  useEffect(async () => {
+    if (connectedUser) {
+      const bondedBalance = await api.call('balanceOf', connectedUser, bondedTokenAddress).toPromise()
+      const daiBalance = await api.call('balanceOf', connectedUser, daiAddress).toPromise()
+      const antBalance = await api.call('balanceOf', connectedUser, antAddress).toPromise()
+      setUserBondedBalance(new BigNumber(bondedBalance))
+      setUserDaiBalance(new BigNumber(daiBalance))
+      setUserAntBalance(new BigNumber(antBalance))
+    }
+  }, [connectedUser])
```

Comment:
Actually I was not sure in the beginning but it didn't complain, I just let it be. Regarding `Promise.all` I don't think it will make them more parallelized, it's just a way to get every promise resolved before moving on right?

Reply:
Also, imagine the `connectedUser` switch to another account while the `getUserBalances` is running.
Since this effect is watching `connectedUser` it will recall `getUSerBalance`. To avoid that you can have a look here: https://github.com/facebook/react/issues/14326#issuecomment-441680293

Reply:
Nope, `Promise.all` IS parallel, not sure it is the case on every browser though

Reply:
I guess it's find if recalls `getUserBalance` since you just switched to another user?
Ah I see what you mean know with `Promise.all`

Reply:
You're right, it should be fine since we need new values.

Reply:
It's the third time I "unresolve" this conversation because change have not been made.
Is it you @deamme marking it as resolved ?

Reply:
@xseignard yes, my bad thought I had pushed my stuff